noetic/distribution.yaml

```diff
@@ -6,7 +6,8 @@ release_platforms:
   debian:
   - buster
   fedora:
-  - '30'
+  - '31'
```

Comment:
If [Fedora 31 EOL is one month after the release of 33](https://fedoraproject.org/wiki/Fedora_Release_Life_Cycle#Maintenance_Schedule) and [Fedora 33 releases 2020/04/08](https://fedorapeople.org/groups/schedule/f-33/f-33-all-tasks.html) then it seems like 31 overlaps with Noetic for less than a month. Can we skip this one?

Reply:
I honestly think that would be best, but we'd be generating RPM spec files based on what we THINK the package names will be. At least for Fedora 32, we have a pretty good idea based on the current state of Rawhide.

Based on the last ~10 or so Fedora releases, I can say with high confidence that the number of differing package names will be extremely small. It's really a matter of policy.

Reply:
* A clarification, on the Fedora 33 schedule you linked, the date you're seeing marks the beginning of Fedora 33 development, not the production release.

Reply:
Fedora 33 G.A. and Fedora 31 EOL is looking more like October 2020: https://fedorapeople.org/groups/schedule/f-33/f-33-key-tasks.html

Reply:
Ah so building RPMs for 31 now will help catch package names that have changed since 30?

Reply:
I think the bigger reason to target Fedora 31 now (pre-release) is to ensure we're generating RPM spec files when we bloom. If we added it to the `distribution.yaml` at release-time, we'd have to re-bloom the world to get specs.

Reply:
Actually, I crossed the streams. I think we should target Fedora 31 now so that we have a production release of Fedora available when Noetic drops, and something stable to test against.

We should target Fedora 32 because we have a good idea what it will look like (based on Rawhide) and because it will be released around the same time Noetic is released.

We historically haven't targeted Fedora versions that aren't even in development yet, so that's why I'm not proposing we target Fedora 33 at this time, though it would fall squarely in the Noetic lifetime.

Reply:
> I think we should target Fedora 31 now so that we have a production release of Fedora available when Noetic drops, and something stable to test against.

[Added Fedora 31 to Rep 3 PR](https://github.com/ros-infrastructure/rep/pull/202/commits/cf4455df07cda7710f3a838168ced367c56e3af8)

Reply:
Thanks. I think the table needs to be updated as well - it's targeting Fedora 30.

Reply:
If we're only going to get 1 month of support out of 31 I suggest that we not target it. The effort to work on it will basically be wasted. And as long as a reasonable prerelease of 32 is available we will get our testing against that.